gh-131116: Fix inspect._finddoc to work with cached_property objects. (#131165)
python/cpython: Doc/library/inspect.rst

   .. versionchanged:: 3.5
      Documentation strings are now inherited if not overridden.

   .. versionchanged:: next

>       Now correctly returns an inherited docstring on :class:`~functools.cached_property` objects if not overridden.

Reviewer: ```suggestion
      Documentation strings on :class:`~functools.cached_property`
      objects are now inherited if not overriden.
```
